src/rgba.rs

```diff
@@ -2,90 +2,155 @@
 // See the COPYRIGHT file at the top-level directory of this distribution.
 // Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
 
-use glib::translate::*;
 use ffi;
-use ffi::GdkRGBA;
-
-pub trait RGBA {
-    fn white() -> GdkRGBA;
-    fn blue() -> GdkRGBA;
-    fn green() -> GdkRGBA;
-    fn red() -> GdkRGBA;
-    fn black() -> GdkRGBA;
-    fn copy(&self) -> GdkRGBA;
-    fn parse(&mut self, spec: &str) -> bool;
-    fn to_string(&self) -> Option<String>;
+use glib::translate::*;
+use glib_ffi::gconstpointer;
+use std::hash::{Hash, Hasher};
+use std::fmt;
+use std::mem;
+use std::str::FromStr;
+
+#[derive(Debug)]
+pub struct RgbaParseError;
+
+#[derive(Clone, Copy, Debug)]
+#[repr(C)]
+pub struct RGBA {
+    pub red: f64,
+    pub green: f64,
+    pub blue: f64,
+    pub alpha: f64,
 }
 
-impl RGBA for GdkRGBA {
-    fn white() -> GdkRGBA {
-        skip_assert_initialized!();
-        GdkRGBA {
-            red: 1f64,
-            green: 1f64,
-            blue: 1f64,
+impl RGBA {
+    pub fn black() -> RGBA {
+        RGBA{
+            red: 0f64,
+            green: 0f64,
+            blue: 0f64,
             alpha: 1f64
         }
     }
 
-    fn blue() -> GdkRGBA {
-        skip_assert_initialized!();
-        GdkRGBA {
+    pub fn blue() -> RGBA {
+        RGBA{
             red: 0f64,
             green: 0f64,
             blue: 1f64,
             alpha: 1f64
         }
     }
 
-    fn green() -> GdkRGBA {
-        skip_assert_initialized!();
-        GdkRGBA {
+    pub fn green() -> RGBA {
+        RGBA{
             red: 0f64,
             green: 1f64,
             blue: 0f64,
             alpha: 1f64
         }
     }
 
-    fn red() -> GdkRGBA {
-        skip_assert_initialized!();
-        GdkRGBA {
+    pub fn red() -> RGBA {
+        RGBA{
             red: 1f64,
             green: 0f64,
             blue: 0f64,
             alpha: 1f64
         }
     }
 
-    fn black() -> GdkRGBA {
-        skip_assert_initialized!();
-        GdkRGBA {
-            red: 0f64,
-            green: 0f64,
-            blue: 0f64,
+    pub fn white() -> RGBA {
+        RGBA{
+            red: 1f64,
+            green: 1f64,
+            blue: 1f64,
             alpha: 1f64
         }
     }
+}
 
-    fn copy(&self) -> GdkRGBA {
-        GdkRGBA {
-            red: self.red,
-            green: self.green,
-            blue: self.blue,
-            alpha: self.alpha
-        }
+impl fmt::Display for RGBA {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let string: String = unsafe {
+            from_glib_full(ffi::gdk_rgba_to_string(self.to_glib_none().0))
+        };
+        f.write_str(&string)
     }
+}
 
-    fn parse(&mut self, spec: &str) -> bool {
+impl Eq for RGBA {}
```

Comment:
Actually, better move `FromStr`, `Hash`, `PartiallyEq` here. This way all xGlibPtrx will go to one place